storm-server/src/main/java/org/apache/storm/scheduler/resource/strategies/scheduling/BaseResourceAwareStrategy.java

```diff
@@ -477,45 +414,136 @@ protected String nodeToRack(RAS_Node node) {
         List<ExecutorDetails> execsScheduled = new LinkedList<>();
 
         Map<String, Queue<ExecutorDetails>> compToExecsToSchedule = new HashMap<>();
-        for (Component component : componentMap.values()) {
-            compToExecsToSchedule.put(component.getId(), new LinkedList<ExecutorDetails>());
+        for (Map.Entry<String, Component> componentEntry: componentMap.entrySet()) {
```

Comment:
Nit we only use the Component out of this and never the key, could we go back to just looping through the values like before?